### PR TITLE
Imposed a barrier on infilling the same follower and leader

### PR DESCRIPTION
**Pull request by Rlamboll**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <https://github.com/znicholls/silicone/pull/XX>`_) Added feature which does something
- (`#XX <https://github.com/znicholls/silicone/pull/XX>`_) Fixed bug identified in (`#YY <https://github.com/znicholls/silicone/issues/YY>`_)
```

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -3,7 +3,8 @@ Changelog
 
 master
 ------
-- (`#27 <https://github.com/znicholls/silicone/pull/27>`_) Developed the constant ratio cruncher
+- (`#29 <https://github.com/znicholls/silicone/pull/29>`_) Prevent asking for infilling with the same  follower and leader variable
```
Comment (znicholls) [inline CHANGELOG.rst]:
```suggestion
- (`#29 <https://github.com/znicholls/silicone/pull/29>`_) Prevent asking for infilling with the same follower and leader variable
```

File: tests/integration/crunchers/test_cruncher_quantile_rolling_windows.py [hunk 2]
```diff
@@ -193,9 +193,27 @@ def test_extreme_values_relationship(self):
...
+            "We can't infill additional follower data from itself".format(
+                variable_follower
+            )
+        )
+        with pytest.raises(ValueError, match=error_msg):
+            res = tcruncher.derive_relationship(variable_follower, [variable_follower])
+            crunched = res(test_db)
+
+    def test_derive_relationship_same_pattern(self, test_db, test_downscale_df):
+        # Given only a single data series, we cannot original pattern
```
Comment (znicholls) [inline tests/integration/crunchers/test_cruncher_quantile_rolling_windows.py]:
```suggestion
        # Given only a single data series, we cannot recreate original pattern
```

??

**Comment by Rlamboll** [comment]
I've realised while away that there is in fact a use-case for leader = follower (using a whole-pathway method to infill times not present in the original data) so I guess we don't want to commit this after all

**Comment by znicholls** [comment]
> I've realised while away that there is in fact a use-case for leader = follower (using a whole-pathway method to infill times not present in the original data) so I guess we don't want to commit this after all

Ah true. It's not a bad strategy to add it now (perhaps raise a `NotImplementedError`) so that you can then unlock it in future once we've got it working. This will prevent people from receiving cryptic errors. So are you happy to re-open and merge and then we make a new PR to close #31 in future?

**Comment by Rlamboll** [comment]
I believe this is already a usable use-case with the RMSClosest cruncher, which at least technically fulfills #31 (in a potentially rather discontinuous way). We can of course work on other, better infillers for this, and should check that this works now.

**Comment by znicholls** [comment]
Ok cool I'll leave #31 open then until you're happy with whatever solution you derive.

**Pull request closed without merging**